connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -118,67 +124,87 @@ class KustoClient(val clusterAlias: String, val engineKcsb: ConnectionStringBuil
     extentsLeftRes.getInt(0) != 0
   }
 
+  def findErrorInResult(res: KustoResultSetTable): (Boolean, Object) = {
+    var error:Object = null
+    var failed = false
+    if (KDSU.getLoggingLevel == Level.DEBUG) {
```

Comment:
Can you consolidate the loops so that debugging is an internal decision?

Reply:
what do you mean by that ?

Reply:
The _if_ and _else_ include share similar code apart for breaking on failures and reporting alerts.
Can you consolidate the code so If Debug is internalized?

You can send him a Jar with this, but it would be better for the official code to include this change